test/dnn/miopen.jl

```diff
@@ -0,0 +1,90 @@
+using AMDGPU.MIOpen
+
+const MIOPEN_TYPES = (
+    (Float16, MIOpen.miopenHalf),
+    (Float32, MIOpen.miopenFloat),
+    (Float64, MIOpen.miopenDouble),
+    (Int8, MIOpen.miopenInt8),
+    (Int32, MIOpen.miopenInt32),
+)
+const MIOPEN_CONV_TYPES = (Float16, Float32) # FIXME: support more types
+
+@testset "Tensor descriptors" begin
+    for nd in 1:MIOpen.MIOPEN_DIM_MAX
+        for (T, mT) in MIOPEN_TYPES
+            x = ROCArray(zeros(T, ntuple(i -> i + 1, Val{nd}())))
+            desc = MIOpen.TensorDescriptor(x)
+            @test ndims(desc) == nd
+
+            dtype, dims, stride = MIOpen.get(desc)
+            @test desc.dtype == mT
+            @test desc.dtype == dtype
+            # NOTE: Dims and strides are reversed to support WHCN convolutions.
+            @test all(dims .== reverse(size(x)))
+            @test all(stride .== reverse(strides(x)))
+        end
+    end
+end
+
+@testset "Simple Convolution" begin
+    for T in MIOPEN_CONV_TYPES, nd in 2:3
+        ndims = Val{nd}()
+        spatial_size = ntuple(i -> 3, ndims)
+        target_size = ntuple(i -> 1, ndims)
+
+        padding = ntuple(i -> 0, ndims)
+        stride = ntuple(i -> 1, ndims)
+        dilation = ntuple(i -> 1, ndims)
+        groups = 1
+
+        x = AMDGPU.ones(T, (spatial_size..., 1, 1))
+        w = AMDGPU.ones(T, (spatial_size..., 1, 1))
+        y = MIOpen.convolution(x, w; padding, stride, dilation, groups)
+        @test eltype(y) == T
+        @test size(y) == (target_size..., 1, 1)
+        @test Array(y)[1] == prod(spatial_size)
+
+        Δ = AMDGPU.ones(T, size(y))
+        ∇w = MIOpen.∇convolution_weight(Δ, x, w; padding, stride, dilation, groups)
+        @test size(∇w) == size(w)
+
+        ∇x = MIOpen.∇convolution_data(Δ, x, w; padding, stride, dilation, groups)
+        @test size(∇x) == size(x)
```

Comment:
Is there any easy way to check that these results are correct? Like how with the rocBLAS tests, we compare against CPU BLAS.

Reply:
Easiest would be to use NNlib. Should we do that?

Reply:
Added!